### DISCUSSION TITLE
Getting a unified docs endpoint for all mounted sub applications

### DISCUSSION THREAD

**Discussion by ollz272** [fastapi/fastapi#10346] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/app")
def read_main():
    return {"message": "Hello World from main app"}


subapi = FastAPI()


@subapi.get("/sub")
def read_sub():
    return {"message": "Hello World from sub API"}


app.mount("/subapi", subapi)
```

### Description

What I want to be able to do is have the /docs endpoint on the main app include docs, with all the associated paths, of the subapplications.

So in the example above, /docs would include docs for both /app, and /subapi/sub

### Operating System

macOS

### FastAPI Version

103.1

### Pydantic Version

2.4.1

### Python Version

3.11.5

**Comment by chrisK824** [comment, marked as answer]
I don't think this is possible based on the docs that declare that those are two fully independent fastapi applications there https://fastapi.tiangolo.com/advanced/sub-applications/?h=mount

That said, maybe you could try and "hack" this thing by messing around with the app instances and their attributes that form the docs with method `openapi`. I wouldn't recommend though even if possible.
